File: Active/Python/P007.py
# Find the 10001st prime number

from is_prime import is_prime 

def compute(nth: int) -> str:
    prime_th: int = 0
    # iterator variable
    i: int = 1

    while prime_th != nth:
        i += 1

        if is_prime(i):
            prime_th += 1

    return i

if __name__ ==  "__main__":
    nth: int = 10001
    print(compute(nth))

File: Active/Java/P007.java
// Project Euler
// Problem 7: The 10001st prime number

public class P007 implements Solution<Integer> {
    public static void main(String[] args) {
        int n_th = 10001;
        System.out.println(new P007().compute(n_th));
    }

    public String compute(Integer n_th) {
        Lib lib = new Lib();

        int prime_th = 0;
        int i = 0;

        for (; prime_th != n_th; i++) {
            if (lib.is_prime(i)) {
                prime_th += 1;
            }
        }

        // minus 1 because, i is getting added before the for loop stop
        return Integer.toString(i - 1);
    }
}
